src/lib/y2network/clients/routing_auto.rb

```diff
@@ -0,0 +1,102 @@
+# Copyright (c) [2019] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "installation/auto_client"
+require "y2network/config"
+require "y2network/routing"
+require "y2network/autoyast_routing"
+require "y2network/routing_helpers"
+
+Yast.import "NetworkInterfaces"
+Yast.import "Lan"
+
+module Y2Network
+  module Clients
+    # This class is reponsible of the autoinstallation routing configuration.
+    class RoutingAuto < ::Installation::AutoClient
+      include RoutingHelpers
+      include Yast::I18n
+      include Yast::Logger
+
+      class << self
+        # @return [Boolean] whether the AutoYaST configuration has been
+        # modified or not
+        attr_accessor :changed
+      end
+
+      # Constructor
+      def initialize
+        textdomain "network"
+      end
+
+      # Import routing configuration
+      #
+      # @param profile [Hash] routing profile section to be imported
+      # @return [Boolean]
+      def import(profile = {})
+        return unless config
```

Comment:
`unless config` looks like typo.

Reply:
Let me check, config was a convenience method which retrieved the current Y2Network::Config, that is Yast::Lan.find_config, so if there was no config then import was not done.

Of course, that is still [WiP], in that case we probably should just create a new one. So open to discussions

Reply:
ah, I though that you want to check that import in not nil. My bad.